src/main/java/org/openrewrite/java/migrate/lang/ThreadStopUnsupported.java

```diff
@@ -0,0 +1,69 @@
+/*
+ * Copyright 2023 the original author or authors.
+ * <p>
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ * <p>
+ * https://www.apache.org/licenses/LICENSE-2.0
+ * <p>
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.openrewrite.java.migrate.lang;
+
+import org.openrewrite.ExecutionContext;
+import org.openrewrite.Preconditions;
+import org.openrewrite.Recipe;
+import org.openrewrite.TreeVisitor;
+import org.openrewrite.java.JavaTemplate;
+import org.openrewrite.java.JavaVisitor;
+import org.openrewrite.java.MethodMatcher;
+import org.openrewrite.java.search.UsesJavaVersion;
+import org.openrewrite.java.tree.J;
+import org.openrewrite.java.tree.TextComment;
+import org.openrewrite.marker.Markers;
+
+import java.util.Collections;
+
+public class ThreadStopUnsupported extends Recipe {
+    private static final MethodMatcher THREAD_STOP_MATCHER = new MethodMatcher("java.lang.Thread stop()");
+
+    @Override
+    public String getDisplayName() {
+        return "Replace `Thread.stop()` with `throw new UnsupportedOperationException()`";
+    }
+
+    @Override
+    public String getDescription() {
+        return "`Thread.stop()` always throws a `new UnsupportedOperationException` in Java 21+. " +
+               "This recipe makes that explicit, as the migration is more complicated." +
+               "See https://docs.oracle.com/en/java/javase/21/docs/api/java.base/java/lang/doc-files/threadPrimitiveDeprecation.html .";
+    }
+
+    @Override
+    public TreeVisitor<?, ExecutionContext> getVisitor() {
+        return Preconditions.check(new UsesJavaVersion<>(21),
```

Comment:
Possibly we would want to do this for earlier releases, too. It was deprecated in Java 1.8 or so.

Reply:
Interesting suggestion.. I'd then lean towards adding the comment for older versions, and only as of Java 21 replace it with that throws, as before that it's still functional. Would you agree with that change?